test/langtools/jdk/javadoc/doclet/testDocletExample/TestDocletExample.java

```diff
@@ -53,18 +54,22 @@ public static void main(String... args) throws Exception {
         t.runTests(m -> new Object[] { Path.of(m.getName()) });
     }
 
-    SnippetUtils snippets = new SnippetUtils("jdk.javadoc");
+    SnippetUtils snippets;
     ToolBox tb = new ToolBox();
 
-    TestDocletExample() {
+    TestDocletExample() throws SnippetUtils.ConfigurationException {
         super(System.out);
+        snippets = new SnippetUtils("jdk.javadoc");
     }
 
     @Test
     public void testEntryPoint(Path base) throws Exception {
         var docletPkg = snippets.getElements().getPackageElement("jdk.javadoc.doclet");
         var dc = snippets.getDocTrees().getDocCommentTree(docletPkg);
         var entryPointSnippet = snippets.getSnippetById(dc, "entry-point");
+        if (entryPointSnippet == null) {
+            throw new Error("Cannot find snippet \"entry-point\"");
```

Comment:
I understand it as follows. Although this code now throws generic `Error` instead of `NullPointerException`, which the bug reporter observed, we shouldn't see that `Error` in circumstances similar to those of bug reporter. This is because in those circumstances the code will throw `ConfigurationException` earlier, at construction time, so we won't reach this check. Do I understand it correctly?

Reply:
Yes. Although we could throw NPE even here, I was wanting to throw something that indicates the test is dysfunctional, as compared to failing or crashing.

Reply:
The rationale that you provided reminds of `new FileInputStream(java.io.File)` that throws `FileNotFoundException`. Unlike the case of `FileInputStream`, not finding a particular snippet is an unexpected, unrecoverable programming error. I cannot imagine a reasonable test that checks that some snippet is not there.

Reply:
It's a reasonable analogy.